R-base is not building with zlib 1.2.11

I can confirm that pinning to `zlib 1.2.8` still works.

Here is the error:

```
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking if zlib version >= 1.2.5... no
configure: error: zlib library and headers are required
checking whether zlib support suffices... Traceback (most recent call last):
```

See for the full log: https://circleci.com/gh/conda-forge/r-base-feedstock/3

xref: https://github.com/conda-forge/zlib-feedstock/pull/12/commits

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@
 language: generic
 
 os: osx
-osx_image: beta-xcode6.1
+osx_image: xcode6.4
 
 env:
   global:
@@ -14,7 +14,7 @@ env:
 
 before_install:
     # Remove homebrew.
-    - brew remove --force $(brew list)
+    - brew remove --force --ignore-dependencies $(brew list)
     - brew cleanup -s
     - rm -rf $(brew --cache)
 
@@ -26,19 +26,14 @@ install:
       bash $MINICONDA_FILE -b
 
       source /Users/travis/miniconda3/bin/activate root
+      conda config --remove channels defaults
+      conda config --add channels defaults
       conda config --add channels conda-forge
       conda config --set show_channel_urls true
       conda install --yes --quiet conda-forge-build-setup
       source run_conda_forge_build_setup
 
-      # install conda-build 2.x to build with a long prefix
-      conda install --yes --quiet conda-build=2
-      conda info
-
 script:
   - conda build ./recipe
 
   - upload_or_check_non_existence ./recipe conda-forge --channel=main
-
-  # inspect the prefix lengths of the built packages
-  - conda inspect prefix-lengths /Users/travis/miniconda3/conda-bld/osx-64/*.tar.bz2

--- a/README.md
+++ b/README.md
@@ -3,7 +3,7 @@ About r-base
 
 Home: http://www.r-project.org/
 
-Package license: GPLv3
+Package license: GPL-3.0
 
 Feedstock license: BSD 3-Clause
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -14,10 +14,6 @@ environment:
     secure: MP4hZYylDyUWEsrt3u3cod2sbFeRwUziH02mvQOdbjsTO/l1yIxDkP/76rSIjcGC
 
   matrix:
-    - TARGET_ARCH: x86
-      CONDA_PY: 27
-      CONDA_INSTALL_LOCN: C:\\Miniconda
-
     - TARGET_ARCH: x64
       CONDA_PY: 27
       CONDA_INSTALL_LOCN: C:\\Miniconda-x64

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -14,6 +14,10 @@ environment:
     secure: MP4hZYylDyUWEsrt3u3cod2sbFeRwUziH02mvQOdbjsTO/l1yIxDkP/76rSIjcGC
 
   matrix:
+    - TARGET_ARCH: x86
+      CONDA_PY: 27
+      CONDA_INSTALL_LOCN: C:\\Miniconda
+
     - TARGET_ARCH: x64
       CONDA_PY: 27
       CONDA_INSTALL_LOCN: C:\\Miniconda-x64
@@ -38,23 +42,19 @@ install:
     # Cywing's git breaks conda-build. (See https://github.com/conda-forge/conda-smithy-feedstock/pull/2.)
     - cmd: rmdir C:\cygwin /s /q
 
-    # Add our channels.
-    - cmd: set "OLDPATH=%PATH%"
-    - cmd: set "PATH=%CONDA_INSTALL_LOCN%\\Scripts;%CONDA_INSTALL_LOCN%\\Library\\bin;%PATH%"
-    - cmd: conda config --set show_channel_urls true
-    - cmd: conda config --add channels conda-forge
-
-    # Add a hack to switch to `conda` version `4.1.12` before activating.
-    # This is required to handle a long path activation issue.
-    # Please see PR ( https://github.com/conda/conda/pull/3349 ).
-    - cmd: conda install --yes --quiet conda=4.1.12
-    - cmd: set "PATH=%OLDPATH%"
-    - cmd: set "OLDPATH="
-
-    # Actually activate `conda`.
+    # Add path, activate `conda` and update conda.
     - cmd: call %CONDA_INSTALL_LOCN%\Scripts\activate.bat
+    - cmd: conda update --yes --quiet conda
+
     - cmd: set PYTHONUNBUFFERED=1
 
+    # Add our channels.
+    - cmd: conda config --set show_channel_urls true
+    - cmd: conda config --remove channels defaults
+    - cmd: conda config --add channels defaults
+    - cmd: conda config --add channels conda-forge
+
+    # Configure the VM.
     - cmd: conda install -n root --quiet --yes obvious-ci
     - cmd: conda install -n root --quiet --yes conda-forge-build-setup
     - cmd: run_conda_forge_build_setup

--- a/ci_support/run_docker_build.sh
+++ b/ci_support/run_docker_build.sh
@@ -14,7 +14,7 @@ config=$(cat <<CONDARC
 
 channels:
  - conda-forge
- - defaults # As we need conda-build
+ - defaults
 
 conda-build:
  root-dir: /feedstock_root/build_artefacts
@@ -25,8 +25,8 @@ CONDARC
 )
 
 cat << EOF | docker run -i \
-                        -v ${RECIPE_ROOT}:/recipe_root \
-                        -v ${FEEDSTOCK_ROOT}:/feedstock_root \
+                        -v "${RECIPE_ROOT}":/recipe_root \
+                        -v "${FEEDSTOCK_ROOT}":/feedstock_root \
                         -a stdin -a stdout -a stderr \
                         condaforge/linux-anvil \
                         bash || exit $?
@@ -48,15 +48,8 @@ source run_conda_forge_build_setup
 # automatically.
 yum install -y mesa-libGL-devel mesa-libGLU-devel libX11-devel libXt-devel libXrender-devel libXext-devel libXdmcp-devel
 
-# install conda-build 2.x to build a long prefix
-conda install --yes --quiet conda-build=2
-conda info
-
 
 # Embarking on 1 case(s).
     conda build /recipe_root --quiet || exit 1
     upload_or_check_non_existence /recipe_root conda-forge --channel=main || exit 1
-
-# inspect the prefix lengths of the built packages
-conda inspect prefix-lengths /feedstock_root/build_artefacts/linux-64/*.tar.bz2
 EOF

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -8,8 +8,6 @@ package:
 source:
   url: https://cran.r-project.org/src/base/R-3/R-{{ version }}.tar.gz
   sha256: d294ad21e9f574fb4828ebb3a94b8cb34f4f304a41687a994be00dd41a4e514c
-  # No openjdk on win32.
-  skip: true  # [win32]
   patches:
     - mac-unicode.patch  # [osx]
     - 0002-Win32-Fix-trio-config-h-include-depth-issue.patch
@@ -29,6 +27,8 @@ build:
     - lib/R/lib/
     - lib/
   detect_binary_files_with_prefix: true
+  # No openjdk on win32.
+  skip: true  # [win32]
 
 requirements:
   build:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,18 +1,15 @@
-# Ported from Continuum Analytics: https://github.com/conda/conda-recipes/tree/master/r-base
-
-{% set name = "r-base" %}
-{% set native = 'm2w64-' if win else '' %}
-{% set posix= 'm2-' if win else '' %}
 {% set version = '3.3.2' %}
-{% set sha1 = "0e39e9c2d28fe6bab9c55ca23e08ba8727fd2fca" %}
+{% set native = 'm2w64-' if win else '' %}
 
 package:
-  name: {{ name|lower }}
+  name: r-base
   version: {{ version }}
 
 source:
   url: https://cran.r-project.org/src/base/R-3/R-{{ version }}.tar.gz
-  sha1: {{ sha1 }}
+  sha256: d294ad21e9f574fb4828ebb3a94b8cb34f4f304a41687a994be00dd41a4e514c
+  # No openjdk on win32.
+  skip: true  # [win32]
   patches:
     - mac-unicode.patch  # [osx]
     - 0002-Win32-Fix-trio-config-h-include-depth-issue.patch
@@ -27,12 +24,11 @@ source:
     - 0012-macOS-include-cairo_h-not-cairo-xlib_h.patch
 
 build:
-  number: 1
+  number: 2
   rpaths:
     - lib/R/lib/
     - lib/
   detect_binary_files_with_prefix: true
-  skip: true  # [win32 or linux32]
 
 requirements:
   build:
@@ -41,67 +37,73 @@ requirements:
     - m2-texinfo               # [win]
     - m2-curl                  # [win]
     - m2-p7zip                 # [win]
-    - readline                 # [not win]
-    - ncurses >=5.9*           # [not win]
+    - readline 6.2*            # [not win]
+    - ncurses 5.9*             # [not win]
     - gcc                      # [not win]
-    - jpeg 8d*                 # [not win]
+    - jpeg 9*                  # [not win]
     - curl                     # [not win]
-    - xz 5.2.*                 # [not win]
     - automake                 # [not win]
-    - pango                    # [linux]
     - openjdk
     - {{native}}pkg-config
-    - {{native}}bzip2          # [not osx]
+    - {{native}}bzip2 1.0.*    # [not osx]
     - {{native}}libjpeg-turbo  # [win]
     - {{native}}toolchain      # [win]
     - {{native}}libiconv       # [win]
     - {{native}}gmp            # [win]
     - {{native}}fftw           # [win]
-    - {{native}}xz             # [win]
+    - {{native}}xz 5.2.*
     - {{native}}mpfr           # [win]
     - {{native}}libsndfile     # [win]
     - {{native}}bwidget        # [win]
     - {{native}}tktable        # [win]
-    - {{native}}pcre           # [win]
+    - {{native}}pcre
+    - {{native}}icu 56.*       # [not win]
     - {{native}}pango 1.40.*   # [not win]
-    - {{native}}pcre 8.39*     # [not win]
-    - {{native}}icu            # [not win]
-    - {{native}}tk
-    - {{native}}zlib
+    - {{native}}tk 8.6.*       # [win]
+    - {{native}}tk 8.5.*       # [not win]
+    # R does not build with zlib 1.2.11
+    # See https://github.com/conda-forge/r-base-feedstock/issues/4
+    - {{native}}zlib 1.2.8
     - {{native}}gsl
-    - {{native}}libpng
-    - {{native}}libtiff
-    - {{native}}libxml2
-
+    - {{native}}libpng >=1.6.21,<1.7  # [win]
+    - {{native}}libpng >=1.6.28,<1.7  # [not win]
+    - {{native}}libtiff 4.0.*
+    - {{native}}libxml2 2.9.*
   run:
-    - pango                    # [linux]
-    - readline                 # [not win]
-    - ncurses >=5.9*           # [not win]
+    - m2-texinfo-tex           # [win]
+    - m2-texinfo               # [win]
+    - m2-curl                  # [win]
+    - m2-p7zip                 # [win]
+    - readline 6.2*            # [not win]
+    - ncurses 5.9*             # [not win]
     - libgcc                   # [not win]
-    - jpeg 8d*                 # [not win]
+    - jpeg 9*                  # [not win]
     - curl                     # [not win]
-    - xz 5.2.*                 # [not win]
-    - {{native}}bzip2          # [not osx]
+    - {{native}}bzip2 1.0.*    # [not osx]
     - {{native}}gcc-libs       # [win]
     - {{native}}libjpeg-turbo  # [win]
+    - {{native}}toolchain      # [win]
     - {{native}}libiconv       # [win]
     - {{native}}gmp            # [win]
     - {{native}}fftw           # [win]
-    - {{native}}xz             # [win]
     - {{native}}mpfr           # [win]
     - {{native}}libsndfile     # [win]
     - {{native}}bwidget        # [win]
     - {{native}}tktable        # [win]
-    - {{native}}pcre           # [win]
+    - {{native}}icu 56.*       # [not win]
     - {{native}}pango 1.40.*   # [not win]
-    - {{native}}pcre 8.39*     # [not win]
-    - {{native}}icu            # [not win]
-    - {{native}}tk
-    - {{native}}zlib
+    - {{native}}tk 8.6.*       # [win]
+    - {{native}}tk 8.5.*       # [not win]
+    - {{native}}libpng >=1.6.21,<1.7  # [win]
+    - {{native}}libpng >=1.6.28,<1.7  # [not win]
+    - {{native}}xz 5.2.*
+    - {{native}}pcre
+    # R does not build with zlib 1.2.11
+    # See https://github.com/conda-forge/r-base-feedstock/issues/4
+    - {{native}}zlib 1.2.8
     - {{native}}gsl
-    - {{native}}libpng
-    - {{native}}libtiff
-    - {{native}}libxml2
+    - {{native}}libtiff 4.0.*
+    - {{native}}libxml2 2.9.*
 
 test:
   commands:
@@ -115,12 +117,16 @@ test:
     - Rfe --help       # [win]
     - Rterm --help     # [win]
     - Rterm --version  # [win]
+    - conda inspect linkages -p $PREFIX r-base  # [not win]
+    - conda inspect objects -p $PREFIX r-base  # [osx]
 
 about:
   home: http://www.r-project.org/
-  license: GPLv3
-  summary: R is a free software environment for statistical computing and graphics.
+  license: GPL-3.0
+  license_file: COPYING
+  summary: 'R is a free software environment for statistical computing and graphics.'
 
 extra:
   recipe-maintainers:
     - johanneskoester
+    - ocefpaf

